make/tools.mk

```diff
@@ -61,36 +61,71 @@ endif
 # --retry-connrefused = retry even if the initial connection was refused
 CURL = curl --silent --show-error --fail --location --retry 10 --retry-connrefused
 
-.PHONY: tools
-tools: $(BINDIR)/tools/helm $(BINDIR)/tools/kubectl $(BINDIR)/tools/kind $(BINDIR)/tools/cosign $(BINDIR)/tools/ginkgo $(BINDIR)/tools/cmrel $(BINDIR)/tools/release-notes $(BINDIR)/tools/controller-gen k8s-codegen-tools $(BINDIR)/tools/goimports $(BINDIR)/tools/go-licenses $(BINDIR)/tools/gotestsum $(BINDIR)/tools/rclone $(BINDIR)/tools/trivy $(BINDIR)/tools/ytt $(BINDIR)/tools/yq
+# In Prow, the pod has the folder "$(BINDIR)/downloaded" mounted into the
+# container. For some reason, even though the permissions are correct,
+# binaries that are mounted with hostPath can't be executed. When in CI, we
+# copy the binaries to work around that. Using $(LN) is only required when
+# dealing with binaries. Other files and folders can be symlinked.
+#
+# Details on how "$(BINDIR)/downloaded" gets cached are available in the
+# description of the PR https://github.com/jetstack/testing/pull/651.
+#
+# We use "printenv CI" instead of just "ifeq ($(CI),)" because otherwise we
+# would get "warning: undefined variable 'CI'".
+ifeq ($(shell printenv CI),)
+LN := ln -f -s
+else
+LN := cp -f -r
+endif
+
+UC = $(shell echo '$1' | tr a-z A-Z)
+LC = $(shell echo '$1' | tr A-Z a-z)
+
+# initialise list of tool names
+TOOL_NAMES := 
+
+define tool_defs
+TOOL_NAMES += $1
+$(call UC,$1)_VERSION ?= $2
+NEEDS_$(call UC,$1) := $$(BINDIR)/tools/$1
+$(call UC,$1) := $$(PWD)/$$(BINDIR)/tools/$1
+
+$$(BINDIR)/tools/$1: $$(BINDIR)/scratch/$(call UC,$1)_VERSION | $$(BINDIR)/downloaded/tools/$1@$$($(call UC,$1)_VERSION)_$$(HOST_OS)_$$(HOST_ARCH) $$(BINDIR)/tools
+	cd $$(dir $$@) && $$(LN) $$(patsubst $$(BINDIR)/%,../%,$$(word 1,$$|)) $$(notdir $$@)
+endef
```

Comment:
Could you document (in a comment above `define`) what `tool_defs` does since it is not obvious at first glance? I.e., explain why we need these two variables.